csharp/LogicalType.cs

```diff
@@ -59,7 +59,7 @@ public override string ToString()
         public static LogicalType Decimal(int precision, int scale = 0) => Create(ExceptionInfo.Return<int, int, IntPtr>(precision, scale, LogicalType_Decimal));
         public static LogicalType Date() => Create(ExceptionInfo.Return<IntPtr>(LogicalType_Date));
         public static LogicalType Time(bool isAdjustedToUtc, TimeUnit timeUnit) => Create(ExceptionInfo.Return<bool, TimeUnit, IntPtr>(isAdjustedToUtc, timeUnit, LogicalType_Time));
-        public static LogicalType Timestamp(bool isAdjustedToUtc, TimeUnit timeUnit) => Create(ExceptionInfo.Return<bool, TimeUnit, IntPtr>(isAdjustedToUtc, timeUnit, LogicalType_Timestamp));
+        public static LogicalType Timestamp(bool isAdjustedToUtc, TimeUnit timeUnit, bool forceSetConvertedType = false) => Create(ExceptionInfo.Return<bool, TimeUnit, bool, IntPtr>(isAdjustedToUtc, timeUnit, forceSetConvertedType, LogicalType_Timestamp));
```

Comment:
```suggestion
        public static LogicalType Timestamp(bool isAdjustedToUtc, TimeUnit timeUnit) => Create(ExceptionInfo.Return<bool, TimeUnit, bool, IntPtr>(isAdjustedToUtc, timeUnit, false, LogicalType_Timestamp));
        public static LogicalType Timestamp(bool isAdjustedToUtc, TimeUnit timeUnit, bool forceSetConvertedType) => Create(ExceptionInfo.Return<bool, TimeUnit, bool, IntPtr>(isAdjustedToUtc, timeUnit, forceSetConvertedType, LogicalType_Timestamp));
```

Hi @Nisden, apologies for asking for another change, but it would be good to avoid changing the ABI here by using an overload instead of adding an optional parameter.

We're working on getting the 8.0.0 release out now so will aim to include this PR.